os/src/ProcessOps.scala

```diff
@@ -251,6 +251,19 @@ case class proc(command: Shellable*) {
   def pipeTo(next: proc): ProcGroup = ProcGroup(Seq(this, next))
 }
 
+object proc {
+
+  /**
+   * The env passed by default to child processes
+   */
+  val env = new scala.util.DynamicVariable[Map[String, String]](sys.env)
+
+  // TODO: Delete when in binary compatibity breaking window
+  def andThen[T](f: proc => T): Seq[Shellable] => T = s => f(apply(s))
+  // TODO: Delete when in binary compatibity breaking window
+  def compose[T](f: T => Seq[Shellable]): T => proc = t => apply(f(t))
```

Comment:
Is this related to this PR? Why is it here?

Reply:
Scala 2 automatically adds `extends AbstractFunctionN` to all case classes with less than 6 (IIRC) arguments.
Here I couldn't find a way to add it since I don't know the type of a Vararg in Scala, or if it even exists. But these methods were created synthetically by Scala 2 and when you add a manual companion object they are missing.
I wanted to create them only for Scala 2 but I think it's fine to added them to Scala 3 as well, since they will be gone when we release os-lib `0.11.0`

Reply:
Not a problem anymore after the change to `os.SubProcess.env`